src/Utils/samples.jl

```diff
@@ -344,6 +344,85 @@ function sample!(D::Vector{VN}, X::LazySet, sampler::CombinedSampler;
     throw(ErrorException("sampling failed"))
 end
 
+"""
+    FaceSampler <: AbstractSampler
+
+Type used for sampling from the `k`-faces of a set.
+
+### Fields
+
+- `dim` -- dimension of the faces to be sampled; a negative number is
+           interpreted as `n - dim` where `n` is the dimension of the set
+
+### Notes
+
+For a three-dimensional polytope, the following face dimensions exist:
+- 3-face – the polytope itself
+- 2-faces – 2-dimensional polygonal faces
+- 1-faces – 1-dimensional edges
+- 0-faces – 0-dimensional vertices
+
+For more information see
+[Wikipedia](https://en.wikipedia.org/wiki/Face_(geometry)#k-face).
+"""
+struct FaceSampler <: AbstractSampler
+    dim::Int
+end
+
+function sample!(D::Vector{VN}, X::LazySet, sampler::FaceSampler;
```

Comment:
```suggestion
function sample!(D::Vector{VN}, X::AbstractHyperrectangle, sampler::FaceSampler;
```

Reply:
in fact i saw that there is a dispatch on hyperrectangles for _sample_faces!, so you can ignore this comment..